Source/DafnyTestGeneration/GenerateTestsCommand.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Linq;
+using Microsoft.Boogie;
+
+namespace Microsoft.Dafny; 
+
+public class GenerateTestsCommand : ICommandSpec {
+  public IEnumerable<IOptionSpec> Options =>
+    new IOptionSpec[] {
+      LoopUnrollOption.Instance,
+      SequenceLengthLimitOption.Instance,
+      TargetMethod.Instance,
+      TestInlineDepth.Instance,
+      VerificationTimeLimitOption.Instance,
+    }.Concat(ICommandSpec.CommonOptions);
+
+  enum Mode {
+    Path,
+    Block
+  }
+
+  readonly Argument<Mode> modeArgument = new("mode", @"
+Block - Prints block-coverage tests for the given program.
+Path - Prints path-coverage tests for the given program.");
+
+  public Command Create() {
+    var result = new Command("generate-tests", "(Experimental) Execute every method in the program that's annotated with the {:test} attribute.");
+    result.AddArgument(modeArgument);
+    result.AddArgument(ICommandSpec.FilesArgument);
+    return result;
+  }
+
+  public void PostProcess(DafnyOptions dafnyOptions, Options options, InvocationContext context) {
+    dafnyOptions.Compile = true;
+    dafnyOptions.RunAfterCompile = false;
+    dafnyOptions.ForceCompile = false;
+    dafnyOptions.CompileVerbose = false;
+    dafnyOptions.ForbidNondeterminism = true;
+    dafnyOptions.DefiniteAssignmentLevel = 2;
+
+    var mode = context.ParseResult.GetValueForArgument(modeArgument);
+    dafnyOptions.TestGenOptions.Mode = mode switch {
+      Mode.Path => TestGenerationOptions.Modes.Path,
+      Mode.Block => TestGenerationOptions.Modes.Block,
+      _ => throw new ArgumentOutOfRangeException()
+    };
+  }
+}
+
+class TargetMethod : StringOption {
+  public static readonly TargetMethod Instance = new();
+  public override object DefaultValue => null!;
```

Comment:
Doesn't `null!` always lead to an exception, or am I misreading this?

Reply:
`!` just tells it not to produce a warning.

Reply:
Ah it's not a "non-null cast" then. It is difficult to make `DefaultValue` an `object?` instead in `IOptionSpec`?